packages/tools/src/stateManagement/segmentation/polySeg/Surface/surfaceComputationStrategies.ts

```diff
@@ -0,0 +1,157 @@
+import { isVolumeSegmentation } from '../../../../tools/segmentation/strategies/utils/stackVolumeCheck';
+import { convertLabelmapToSurface } from './convertLabelmapToSurface';
+
+export type RawSurfacesData = {
+  segmentIndex: number;
+  data: Types.SurfaceData;
+}[];
+
+/**
+ * Computes surface data for a given segmentation.
+ * @param segmentationId - The ID of the segmentation.
+ * @param options - Additional options for surface computation.
+ * @returns A promise that resolves to the computed surface data.
+ * @throws An error if there is no surface data available for the segmentation.
+ */
+export async function computeSurfaceData(
+  segmentationId: string,
+  options: PolySegConversionOptions = {}
+) {
+  const segmentIndices = options.segmentIndices?.length
+    ? options.segmentIndices
+    : getUniqueSegmentIndices(segmentationId);
+
+  let rawSurfacesData: RawSurfacesData;
+  const segmentation = getSegmentation(segmentationId);
+  const representationData = segmentation.representationData;
+
+  try {
+    if (representationData.CONTOUR) {
+      rawSurfacesData = await computeSurfaceFromContourSegmentation(
+        segmentationId,
+        {
+          segmentIndices,
+          ...options,
+        }
+      );
+    } else if (representationData.LABELMAP as LabelmapSegmentationData) {
+      // convert volume labelmap to surface
+      rawSurfacesData = await computeSurfaceFromLabelmapSegmentation(
+        segmentation.segmentationId,
+        {
+          segmentIndices,
+          ...options,
+        }
+      );
+    }
+  } catch (error) {
+    console.error(error);
+    throw error;
+  }
+
+  if (!rawSurfacesData) {
+    throw new Error(
+      'Not enough data to convert to surface, currently only support converting volume labelmap to surface if available'
+    );
+  }
+
+  const surfacesData = await createAndCacheSurfacesFromRaw(
+    segmentationId,
+    rawSurfacesData,
+    options
+  );
+
+  return surfacesData;
+}
+
+async function computeSurfaceFromLabelmapSegmentation(
+  segmentationId,
+  options: PolySegConversionOptions = {}
+): Promise<RawSurfacesData> {
+  // Todo: validate valid labelmap representation
+  const segmentation = getSegmentation(segmentationId);
+
+  if (!segmentation?.representationData?.LABELMAP) {
+    console.warn('Only support surface update from labelmaps');
+    return;
+  }
+
+  const isVolume = isVolumeSegmentation(
+    segmentation.representationData.LABELMAP
+  );
+
+  const labelmapRepresentationData = segmentation.representationData.LABELMAP;
+
+  const segmentIndices =
+    options.segmentIndices || getUniqueSegmentIndices(segmentationId);
+
+  const promises = segmentIndices.map((index) => {
+    const surface = convertLabelmapToSurface(
+      labelmapRepresentationData as
+        | LabelmapSegmentationDataVolume
+        | LabelmapSegmentationDataStack,
+      index,
+      isVolume
+    );
+
+    return surface;
+  });
+
+  const surfaces = await Promise.all(promises);
```

Comment:
Should this allow reporting individual failures when they occur, using alLSettled perhaps, and then mapping to an error issue below?